programs/fileio.c

```diff
@@ -862,6 +862,193 @@ int FIO_compressFilename(const char* dstFileName, const char* srcFileName,
+                exit(1);
+            }
+            frameSize = MEM_readLE32(frameSizeBuffer);
+            fseek(srcFile, frameSize, SEEK_CUR);
+        }
+
+    }
+    return 0;
+}
+void displayInfo(const char* inFileName, fileInfo_t* info, int displayLevel){
+    double compressedSizeMB = (double)info->compressedSize/(1 MB);
+    double decompressedSizeMB = (double)info->decompressedSize/(1 MB);
+
+    if(displayLevel<=2){
+        if(info->usesCheck && info->canComputeDecompSize){
+            DISPLAY("Skippable  Non-Skippable  Compressed  Uncompressed  Ratio  Check  Filename\n");
+            DISPLAY("%9d  %13d  %7.2f MB    %7.2f MB  %5.3f  XXH64  %s\n",
+                    info->numSkippableFrames, info->numActualFrames, compressedSizeMB, decompressedSizeMB,
+                    compressedSizeMB/decompressedSizeMB, inFileName);
+        }
+        else if(!info->usesCheck){
+            DISPLAY("Skippable  Non-Skippable  Compressed  Uncompressed  Ratio  Check  Filename\n");
+            DISPLAY("%9d  %13d  %7.2f MB    %7.2f MB  %5.3f  %s\n",
+                    info->numSkippableFrames, info->numActualFrames, compressedSizeMB, decompressedSizeMB,
+                    compressedSizeMB/decompressedSizeMB, inFileName);
+        }
+        else if(!info->canComputeDecompSize){
+            DISPLAY("Skippable  Non-Skippable  Compressed  Uncompressed  Ratio  Check  Filename\n");
+            DISPLAY("%9d  %13d  %7.2f MB      XXH64  %s\n",
+                    info->numSkippableFrames, info->numActualFrames, compressedSizeMB, inFileName);
+        }
+        else{
+            DISPLAY("Skippable  Non-Skippable  Filename\n");
+            DISPLAY("%9d  %13d  %7.2f MB  %s\n",
+                    info->numSkippableFrames, info->numActualFrames, compressedSizeMB, inFileName);
+        }
+    }
+    else{
+        DISPLAY("# Zstandard Frames: %d\n", info->numActualFrames);
+        DISPLAY("# Skippable Frames: %d\n", info->numSkippableFrames);
+        DISPLAY("Compressed Size: %.2f MB (%llu B)\n", compressedSizeMB, info->compressedSize);
+        if(info->canComputeDecompSize){
+            DISPLAY("Decompressed Size: %.2f MB (%llu B)\n", decompressedSizeMB, info->decompressedSize);
+            DISPLAY("Ratio: %.4f\n", compressedSizeMB/decompressedSizeMB);
+        }
+        if(info->usesCheck){
+            DISPLAY("Check: XXH64\n");
+        }
+    }
+
+}
+
+int FIO_listFile(const char* inFileName, int displayLevel){
+    const char* const suffixPtr = strrchr(inFileName, '.');
+    DISPLAY("File: %s\n", inFileName);
+    if(!suffixPtr || strcmp(suffixPtr, ZSTD_EXTENSION)){
+        DISPLAYLEVEL(1, "file %s was not compressed with zstd -- ignoring\n\n", inFileName);
+        return 1;
+    }
+    else{
+        fileInfo_t* info = (fileInfo_t*)malloc(sizeof(fileInfo_t));
+        int error = getFileInfo(info, inFileName);
+        if(error==1){
+            DISPLAY("An error occurred with getting file info\n");
+            exit(1);
```

Comment:
Better signal an error, but continue processing.
Ex : 
```
xz -l LICENSE  xxhsum.c.xz
xz: LICENSE: File format not recognized
Strms  Blocks   Compressed Uncompressed  Ratio  Check   Filename
    1       1       9988 B     38.8 KiB  0.251  CRC64   xxhsum.c.xz
echo $?
1
```

The program end result is still an error if one file fails.